apps/frontend/src/pages/settings/profile.vue

```diff
@@ -160,16 +165,30 @@ function showPreviewImage(files) {
   };
 }
 
+function removePreviewImage() {
+  pendingAvatarDeletion.value = true;
+  previewImage.value = "https://cdn.modrinth.com/placeholder.png";
+}
+
 function cancel() {
   icon.value = null;
   previewImage.value = null;
+  pendingAvatarDeletion.value = false;
   username.value = auth.value.user.username;
   bio.value = auth.value.user.bio;
 }
 
 async function saveChanges() {
   startLoading();
   try {
+    if (pendingAvatarDeletion.value) {
+      await useBaseFetch(`user/${auth.value.user.id}/icon`, {
+        method: "DELETE",
+      });
```

Comment:
From my understanding this wouldn't work without adding the `apiVersion` field.

```suggestion
      await useBaseFetch(`user/${auth.value.user.id}/icon`, {
        method: "DELETE",
        apiVersion: 3
      });
```